core/src/main/java/org/apache/iceberg/avro/BaseWriteBuilder.java

```diff
@@ -64,11 +64,6 @@ public ValueWriter<?> map(Schema map, ValueWriter<?> valueWriter) {
     return ValueWriters.map(ValueWriters.strings(), valueWriter);
   }
 
-  @Override
-  public ValueWriter<?> variant(Schema variant, List<ValueWriter<?>> fields) {
```

Comment:
I reverted this for now and will implement after https://github.com/apache/iceberg/commit/125fa6a56d6b6d741d1bc9c5cf9fe94825ffa44a.